src/main/java/com/shippo/model/Batch.java

```diff
@@ -0,0 +1,357 @@
+import com.shippo.net.APIResource;
+
+/*
+ * Represents ../batches endpoint documented at https://goshippo.com/docs/reference#batches
+ */
+public final class Batch extends APIResource {
+
+	@SerializedName("object_id")
+	private String id;
+
+	@SerializedName("object_owner")
+	private String owner;
+
+	public static enum BatchStatus {
+		VALIDATING, VALID, INVALID, PURCHASING, PURCHASED
+	}
+
+	@SerializedName("object_status")
+	private BatchStatus status;
+
+	@SerializedName("object_created")
+	private Date created;
+
+	@SerializedName("object_updated")
+	private Date updated;
+	private String metadata;
+	private String defaultCarrierAccount;
+	private String defaultServiceLevelToken;
+	private LabelFileType labelFileType;
+
+	public static class BatchShipmentCollection {
+		private String next;
+		private String previous;
+
+		@SerializedName("results")
+		private BatchShipment[] shipments;
+
+		@Override
+		public String toString() {
+			return "BatchShipmentCollection [next=" + next + ", shipments=" + Arrays.toString(shipments) + "]";
+		}
+
+		public String getNext() {
+			return next;
+		}
+
+		public String getPrevious() {
+			return previous;
+		}
+
+		public BatchShipment[] getShipments() {
+			return shipments;
+		}
+	}
+
+	private BatchShipmentCollection batchShipments;
+
+	@SerializedName("label_url")
+	private String[] labelURLs;
+
+	public static class Counts {
+		private int creationSucceeded;
+		private int creationFailed;
+		private int purchaseSucceeded;
+		private int purchaseFailed;
+
+		@Override
+		public String toString() {
+			return "Counts [creationSucceeded=" + creationSucceeded + ", creationFailed=" + creationFailed
+					+ ", purchaseSucceeded=" + purchaseSucceeded + ", purchaseFailed=" + purchaseFailed + "]";
+		}
+
+		public int getCreationSucceeded() {
+			return creationSucceeded;
+		}
+
+		public int getCreationFailed() {
+			return creationFailed;
+		}
+
+		public int getPurchaseSucceeded() {
+			return purchaseSucceeded;
+		}
+
+		public int getPurchaseFailed() {
+			return purchaseFailed;
+		}
+	}
+
+	private Counts objectResults;
+
+	@Override
+	public String toString() {
+		return "Batch [id=" + id + ", owner=" + owner + ", status=" + status + ", created=" + created + ", updated="
+				+ updated + ", metadata=" + metadata + ", defaultCarrierAccount=" + defaultCarrierAccount
+				+ ", defaultServiceLevelToken=" + defaultServiceLevelToken + ", labelFileType=" + labelFileType
+				+ ", batchShipments=" + batchShipments + ", labelURLs=" + Arrays.toString(labelURLs)
+				+ ", objectResults=" + objectResults + "]";
+	}
+
+	private static class BatchCollection {
+		private int count;
+		private String previous;
+		private String next;
+		@SerializedName("results")
+		private Batch[] array;
+	}
+
+	// Taken from
+	// http://stackoverflow.com/questions/13592236/parse-a-uri-string-into-name-value-collection
+	private static Map<String, String> queryToParams(String query) throws UnsupportedEncodingException {
```

Comment:
Thank you for this suggestion to guide in the right direction and reminding me how lazy I am. It happens all one wants to do is *FINISH* it :)